src/category_theory/limits/presheaf.lean

```diff
@@ -312,10 +312,27 @@ nat_iso_of_nat_iso_on_representables _ _ (hL ≪≫ (is_extension_along_yoneda _
 If `L` preserves colimits and `ℰ` has them, then it is a left adjoint. Note this is a (partial)
 converse to `left_adjoint_preserves_colimits`.
 -/
-def is_left_adjoint_of_preserves_colimits (L : (Cᵒᵖ ⥤ Type u₁) ⥤ ℰ) [preserves_colimits L] :
+def is_left_adjoint_of_preserves_colimits (L : (C ⥤ Type u₁) ⥤ ℰ) [preserves_colimits L] :
   is_left_adjoint L :=
-{ right := restricted_yoneda (yoneda ⋙ L),
-  adj := (yoneda_adjunction _).of_nat_iso_left
-              (unique_extension_along_yoneda _ L (iso.refl _)).symm }
+begin
```

Comment:
This used to be a short definition in term mode, now it's long and in tactic mode. Will this create issues with defeqs down the road? Can you show that it behaves as expected with some simp-lemmas proven by `rfl`?

Reply:
This is a fair concern - I think I'm not particularly worried about it though because the right adjoint would be unique up to isomorphism anyway, and I can't immediately see a need for explicit usage of the definition.

Could you suggest which simp-lemmas you'd expect to see here?

Reply:
I haven't figured out what the simp-lemma should look like. But from the current code, we can't ever unfold the definition and get something useful. Maybe we don't care, like you point out.
I'll let you decide.

Reply:
Perhaps the sensible approach is to change the `have`s to `let`s, and add a todo saying to consider adding in lemmas characterising the adjoint (basically the same as the discussion and outcome here: https://github.com/leanprover-community/mathlib/pull/3924)

Reply:
Is this version alright?